lightning/src/ln/channelmonitor.rs

```diff
@@ -966,6 +982,15 @@ impl<ChanSigner: ChannelKeys + Writeable> ChannelMonitor<ChanSigner> {
 			}
 		}
 
+		(self.watch_outputs.len() as u64).write(writer)?;
```

Comment:
this is quite the write function. It might be prudent to break it out into subroutines so it's easier to navigate.

Reply:
Agreed. We should probably implement serialize on subtypes so that we can just call the write derive macro, but its not particularly worse here, just still-bad.